simple_use_case/aiida/simple_use_case.py

```diff
@@ -0,0 +1,96 @@
+from pathlib import Path
+
+from aiida.orm import List, SinglefileData
+from aiida.engine.processes.functions import shellfunction, workfunction
+
+
+@shellfunction(command="gmsh", output_filenames=["mesh"])
+def gmsh(**kwargs):
+    """Run the ``gmsh`` command."""
+
+
+@shellfunction(command="meshio", output_filenames=["mesh.xdmf", "mesh.h5"])
+def meshio(**kwargs):
+    """Run the ``meshio`` command to convert to XDMF format."""
+
+
+@shellfunction(command="python", output_filenames=["*.pvd", "*.vtu"])
+def poisson(**kwargs):
+    """Run the ``poisson`` step."""
+
+
+@shellfunction(command="pvbatch", output_filenames=["plotoverline.csv"])
+def pvbatch(**kwargs):
+    """Run the ``pvbatch`` command."""
+
+
+@workfunction
+def workflow(geometry, poisson_script, post_processing_script):
+    """Simple workflow to solve Poisson equation for a geometry."""
+
+    # Generate the mesh from the geometry.
+    # Instead of writing to "mesh.msh" which would allow gmsh to deduce the output format
+    # we explicitly specify the format using the "-format" option.
+    # For certain reasons, the output link label in AiiDA would be converted to "mesh_msh"
+    # if we would have used "mesh.msh".
+    # For more information see section "Design choices" > "Output files" in the AiiDA
+    # Enhancement Proposal https://github.com/aiidateam/AEP/tree/aep/shell-functions/xxx_shell_functions
+    arguments = List(
+        ["-2", "-order", "1", "-format", "msh2", "{geometry}", "-o", "mesh"]
+    )
+    results_gmsh = gmsh(arguments=arguments, geometry=geometry)
+
+    # Convert the mesh to XDMF format
+    # The mesh file generated by gmsh is written to a temporary working directory
+    # with a temporary filename. Therefore, meshio can't deduce the input format
+    # from the filename and we have to specify this explicitly.
+    arguments = List(["convert", "--input-format", "gmsh", "{mesh}", "mesh.xdmf"])
+    results_meshio = meshio(arguments=arguments, mesh=results_gmsh["mesh"])
+
+    # Solve poisson equation
+    # The additional argument '--h5' is necessary to specify the .h5 file as input,
```

Comment:
I don't understand this, what is that temp directory?

Reply:
The shell command is executed in some temporary directory set as the working directory. See here [code](https://github.com/sphuber/aiida-core/blob/feature/5287/shell-function/aiida/engine/processes/functions/shell.py#L156-L189)

Reply:
So in an HPC context the working directory would be the HPC machine that needs to have all the inputs and copy the outputs back. So isn't that then expected? I guess this is somehow confusing with the statement that the shell_function can only be used on a local machine, but maybe the general way of handling inputs/outputs is similar. So I would just write working directory, and the file name IMO is not temporary (the file in the working directory is but not the name).